.gitignore

```diff
@@ -22,6 +22,16 @@ yarn-error.log
 .eslintcache
 .vscode
 
+# yarn
+# https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored
+.pnp.*
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```

Comment:
Why would we not ignore these?

Reply:
I copied this list straight from the yarn docs here: https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored
There's an explanation in that FAQ entry for each path.

Reply:
Regarding checking in yarn sources, afaics we have two options when using yarn v2+:
* use the experimental [corepack](https://yarnpkg.com/corepack) feature
* or commit the yarn sources

see https://github.com/yarnpkg/berry/issues/5748 for more details

Reply:
`corepack` seems preferable, I am strongly against vendoring yarn code in the repo (feels insane to me that they would even propose this as a default setup).

In the issue there were mentions of problems with how corepack operates depending on versions. Probably not an issue for jaeger-ui.

Q: why yarn 3.8 and not 4.x?

Reply:
The reason for yarn 3.8 was that we want to use cachi2 (used to ensure the build is hermetic, i.e. it caches dependencies to support building Jaeger UI without network access), which at the moment [only supports yarn v3](https://github.com/containerbuildsystem/cachi2/blob/main/docs/yarn.md) (I assume v4 might also be supported, but I didn't test it).

Alternatively, we could also use `npm`. It also supports workspaces now (in case this was the reason to switch to yarn in the first place). What do you think?

Reply:
I don't really follow. We have a yarn.lock file that ensures repeatable build. Are we worried about supply chain attack?

Reply:
Yes, it's about supply chain attacks. Our downstream builds need to be [hermetic according to SLSA](https://slsa.dev/spec/v0.1/requirements#hermetic) and include a Software Bill of Materials (SBOM) etc.

You're right, effectively the yarn package manager already supports these properties. We need to support multiple languages and package managers though, so this part of the build pipeline is handled by an external tool like [cachi2](https://github.com/containerbuildsystem/cachi2) for our downstream builds.